presto-tests/src/main/java/com/facebook/presto/tests/sql/TestTable.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.sql;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestTable
+        implements AutoCloseable
+{
+    private final SqlExecutor sqlExecutor;
+    private final String name;
+
+    private static final AtomicInteger instanceCounter = new AtomicInteger();
+
+    public TestTable(SqlExecutor sqlExecutor, String namePrefix, String createDdlTemplate)
+    {
+        this.sqlExecutor = sqlExecutor;
+        this.name = namePrefix + "_" + instanceCounter.incrementAndGet();
+        sqlExecutor.execute(createDdlTemplate.replace("{TABLE_NAME}", this.name));
```

Comment:
I forgot there's a more important reason for templating: we can't really DROP the table if we don't know with what name it was created in the first place. If we take a `ddl` instead of `ddlTemplate`, and take the `name` from the caller, we have no guarantee that it was created with the name provided. Also, this would require the caller to repeat themselves:

`new TestTable(sqlExecutor, "NATION", "CREATE TABLE NATION AS ...")`

We could try parsing the name, doing another sql call to verify table creation (costly), or kill two birds with one stone and have unique names gratis:

`new TestTable(sqlExecutor, "tpch.name_prefix", "CREATE TABLE {TABLE_NAME} AS ...")`